packages/flutter/lib/src/rendering/grid.dart

```diff
@@ -73,65 +273,78 @@ class RenderGrid extends RenderBox with ContainerRenderObjectMixin<RenderBox, Gr
       child.parentData = new GridParentData();
   }
 
+  Size _getIntrinsicSize(BoxConstraints constraints) {
+    return constraints.constrain(_delegate.getGridSpecification(constraints, childCount).gridSize);
+  }
+
   double getMinIntrinsicWidth(BoxConstraints constraints) {
     assert(constraints.isNormalized);
-    // We can render at any width.
-    return constraints.constrainWidth(0.0);
+    return _getIntrinsicSize(constraints).width;
   }
 
   double getMaxIntrinsicWidth(BoxConstraints constraints) {
     assert(constraints.isNormalized);
-    double maxWidth = childCount * _maxChildExtent;
-    return constraints.constrainWidth(maxWidth);
+    return _getIntrinsicSize(constraints).width;
```

Comment:
You're not allowed to size things around text with this render object.  You have to commit to the grid specification (which determines the size of the grid) before you're allowed to look at the children.  In fact, you're never allowed to see the children.  All you can see is the index of the child and it's placementData.